spyder/utils/programs.py

```diff
@@ -138,6 +139,19 @@ def alter_subprocess_kwargs_by_platform(**kwargs):
         # We "or" them together

```

Comment:
Also this `alter_subprocess_kwargs_by_platform``

We could add a test to check that the expected values on windows are there etc.